### ISSUE TITLE
commit d759a7ce break ListBuckets and ListObjects on FUSE drive 

### ISSUE THREAD

**Issue by joriscarrier** [minio/minio#8252]
Hello, with this commit [d759a7ce](https://github.com/minio/minio/commit/d759a7ce99d1d03774ac005178480bcd717d299a)
ListBuckets & ListObjects return nothing, I don't understand why (tested with fuse & fuse3)

can be reproduce with python3 or C exemple:

python3 (with fusepy) & fuse 2.9.9-1 [loopback.py](https://raw.githubusercontent.com/fusepy/fusepy/master/examples/loopback.py)
C & fuse 3.6.2 [passthrough.c](https://github.com/libfuse/libfuse/blob/master/example/passthrough.c)

logs with python3 & fuse 

> without commit d759a7ce

```
DEBUG:fuse:-> opendir /root/ ()
DEBUG:fuse:<- opendir 0
DEBUG:fuse:-> readdir /root/ (0,)
DEBUG:fuse:<- readdir ['.', '..', '.minio.sys', 'test']
DEBUG:fuse:-> getattr /root/.minio.sys (None,)
DEBUG:fuse:<- getattr {'st_atime': 1568663371.498455, 'st_ctime': 1568663371.538455, 'st_gid': 0, 'st_mode': 16877, 'st_mtime': 1568663371.538455, 'st_nlink': 5, 'st_size': 4096, 'st_uid': 0}
DEBUG:fuse:-> getattr /root/test (None,)
DEBUG:fuse:<- getattr {'st_atime': 1568663404.14736, 'st_ctime': 1568663404.14736, 'st_gid': 0, 'st_mode': 16877, 'st_mtime': 1568663404.14736, 'st_nlink': 2, 'st_size': 4096, 'st_uid': 0}
DEBUG:fuse:-> releasedir /root/ (0,)
DEBUG:fuse:<- releasedir 0
DEBUG:fuse:-> opendir /root/ ()
DEBUG:fuse:<- opendir 0
DEBUG:fuse:-> readdir /root/ (0,)
DEBUG:fuse:<- readdir ['.', '..', '.minio.sys', 'test']
DEBUG:fuse:-> releasedir /root/ (0,)
DEBUG:fuse:<- releasedir 0
```

> with commit d759a7ce

```
DEBUG:fuse:-> opendir /root/ ()
DEBUG:fuse:<- opendir 0
DEBUG:fuse:-> readdir /root/ (0,)
DEBUG:fuse:<- readdir ['.', '..', '.minio.sys', 'test']
DEBUG:fuse:-> releasedir /root/ (0,)
DEBUG:fuse:<- releasedir 0
DEBUG:fuse:-> opendir /root/ ()
DEBUG:fuse:<- opendir 0
DEBUG:fuse:-> readdir /root/ (0,)
DEBUG:fuse:<- readdir ['.', '..', '.minio.sys', 'test']
DEBUG:fuse:-> releasedir /root/ (0,)
DEBUG:fuse:<- releasedir 0
```

**Comment by harshavardhana** [comment]
@joriscarrier - I don't know what this is and what is the real use-case here?  can you explain this further? 

Setting this as won't fix for now.

**Comment by joriscarrier** [comment]
I use a fuse mountpoint (for example /mnt/fuse) in which I run minio (`minio server /mnt/fuse/minio`) when I want to list objects or buckers (with the command `mc ls` same effect with boto3) the command returns nothing (the response of minio is 200 OK (seen with mc admin trace) when I recompile minio without the commit d759a7ce the command `mc ls` works perfectly

**Comment by vadmeste** [comment]
@joriscarrier Can you paste the failed mc command with --debug flag enabled ?

**Comment by joriscarrier** [comment]
> failed
```root@73fd85d7b5af:/usr/src/mc# mc ls myminio --debug
mc: <DEBUG> GET / HTTP/1.1
Host: 172.17.0.2:9000
User-Agent: MinIO (linux; amd64) minio-go/v6.0.32 mc/2019-09-16T19:38:37Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20190917/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20190917T141758Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 270
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Date: Tue, 17 Sep 2019 14:17:58 GMT
Server: MinIO/DEVELOPMENT.2019-09-17T14-17-45Z
Vary: Origin
X-Amz-Request-Id: 15C53F833BD35C38
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  4.3478ms

mc: <DEBUG> GET / HTTP/1.1
Host: 172.17.0.2:9000
User-Agent: MinIO (linux; amd64) minio-go/v6.0.32 mc/2019-09-16T19:38:37Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20190917/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20190917T141758Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 270
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Date: Tue, 17 Sep 2019 14:17:58 GMT
Server: MinIO/DEVELOPMENT.2019-09-17T14-17-45Z
Vary: Origin
X-Amz-Request-Id: 15C53F833C1B59E8
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  2.2519ms
```


> success

```
root@73fd85d7b5af:/usr/src/mc# mc ls myminio --debug
mc: <DEBUG> GET / HTTP/1.1
Host: 172.17.0.2:9000
User-Agent: MinIO (linux; amd64) minio-go/v6.0.32 mc/2019-09-16T19:38:37Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20190917/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20190917T141656Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 357
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Date: Tue, 17 Sep 2019 14:16:56 GMT
Server: MinIO/DEVELOPMENT.2019-09-17T11-15-19Z
Vary: Origin
X-Amz-Request-Id: 15C53F74F3F4D878
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  6.1896ms

mc: <DEBUG> GET / HTTP/1.1
Host: 172.17.0.2:9000
User-Agent: MinIO (linux; amd64) minio-go/v6.0.32 mc/2019-09-16T19:38:37Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20190917/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20190917T141656Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 357
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Date: Tue, 17 Sep 2019 14:16:56 GMT
Server: MinIO/DEVELOPMENT.2019-09-17T11-15-19Z
Vary: Origin
X-Amz-Request-Id: 15C53F74F45C7A28
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  3.7765ms

[2019-09-16 22:52:02 CEST]      0B test/
```

**Comment by harshavardhana** [comment]
Are you saying that you are exporting a fuse mount to MinIO ? The reason it is skipped is because #8240

**Comment by harshavardhana** [comment]
This is fixed in master in PR #8240
